instrumentation/github.com/go-gorm/gorm/otelgorm/plugin_test.go

```diff
@@ -0,0 +1,241 @@
+
+func closeDB(db *gorm.DB) {
+	sqlDB, err := db.DB()
+	if err != nil {
+		sqlDB.Close()
+	}
+}
+
+func TestPlugin(t *testing.T) {
+
+	testCases := []struct {
+		name           string
+		testOp         func(db *gorm.DB) *gorm.DB
+		spans          int
+		targetSpan     int
+		expectSpanName string
+		sqlOp          string
+		affectedRows   int64
+	}{
+		{
+			"create (insert) row",
+			func(db *gorm.DB) *gorm.DB {
+				return db.Create(&TestModel{Code: "D42", Price: 100})
+			},
+			2,
+			0,
+			"INSERT db.test_models",
+			"INSERT",
+			1,
+		},
+		{
+			"save (update) row",
+			func(db *gorm.DB) *gorm.DB {
+				tm := TestModel{ID: 1, Code: "D42", Price: 100}
+				db = db.Create(&tm)
+				if db.Error != nil {
+					return db
+				}
+				tm.Code = "foo"
+				return db.Save(&tm)
+			},
+			3,
+			1,
+			"UPDATE db.test_models",
+			"UPDATE",
+			1,
+		},
+		{
+			"delete row",
+			func(db *gorm.DB) *gorm.DB {
+				tm := TestModel{ID: 1, Code: "D42", Price: 100}
+				db = db.Create(&tm)
+				if db.Error != nil {
+					return db
+				}
+				return db.Delete(&tm)
+			},
+			3,
+			1,
+			"DELETE db.test_models",
+			"DELETE",
+			1,
+		},
+		{
+			"query row",
+			func(db *gorm.DB) *gorm.DB {
+				tm := TestModel{ID: 1, Code: "D42", Price: 100}
+				db = db.Create(&tm)
+				if db.Error != nil {
+					return db
+				}
+				return db.First(&tm)
+			},
+			3,
+			1,
+			"SELECT db.test_models",
+			"SELECT",
+			1,
+		},
+		{
+			"raw",
+			func(db *gorm.DB) *gorm.DB {
+				tm := TestModel{ID: 1, Code: "D42", Price: 100}
+				db = db.Create(&tm)
+				if db.Error != nil {
+					return db
+				}
+
+				var result []TestModel
+				return db.Raw("SELECT * FROM test_models").Scan(&result)
+			},
+			3,
+			1,
+			"SELECT db.test_models",
+			"SELECT",
+			-1,
+		},
+		{
+			"row",
+			func(db *gorm.DB) *gorm.DB {
+				tm := TestModel{ID: 1, Code: "D42", Price: 100}
+				db = db.Create(&tm)
+				if db.Error != nil {
+					return db
+				}
+
+				db.Raw("SELECT id FROM test_models").Row()
+				return &gorm.DB{Error: nil}
+			},
+			3,
+			1,
+			"SELECT db.test_models",
+			"SELECT",
+			-1,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(tt *testing.T) {
+			db, err := initDB()
+			defer closeDB(db)
+
+			assert.NoError(tt, err)
+
+			sr := new(oteltest.SpanRecorder)
+			provider := oteltest.NewTracerProvider(oteltest.WithSpanRecorder(sr))
+
+			plugin := NewPlugin(WithTracerProvider(provider), WithDBName("db"))
+
+			err = db.Use(plugin)
+			assert.NoError(tt, err)
+
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+			defer cancel()
+
+			ctx, span := provider.Tracer(defaultTracerName).Start(ctx, "gorm-test")
+
+			db = db.WithContext(ctx)
+			// Create
+			dbOp := tc.testOp(db)
+			assert.NoError(tt, dbOp.Error)
+
+			span.End()
+
+			spans := sr.Completed()
+			require.Len(t, spans, tc.spans)
+			s := spans[tc.targetSpan]
+
+			assert.Equal(tt, spans[0].SpanContext().TraceID, spans[1].SpanContext().TraceID)
```

Comment:
bug
```suggestion
			assert.Equal(tt, spans[0].SpanContext().TraceID(), spans[1].SpanContext().TraceID())
```

nit (maybe subjective): I like adding explicit assertion messages explaining why this assertion even exists like `should record spans under the same trace` - it just helps me understanding the tests quicker and also getting quicker insight when something fails

Reply:
thats a good tip thanks